src/Notes/WC_Admin_Notes.php

```diff
@@ -100,6 +103,73 @@ public static function delete_notes_with_name( $name ) {
 		}
 	}
 
+	/**
+	 * Update a note.
+	 *
+	 * @param WC_Admin_Note $note The note that will be updated.
+	 * @param array         $requested_updates a list of requested updates.
+	 */
+	public static function update_note( $note, $requested_updates ) {
```

Comment:
This is potentially confusing, I would expect this to be able to handle any updates I throw at it - ie it should be an exhaustive list of properties that would then need to be maintained. However it can only handle these three properties. I feel like the code that calls this should be handling this logic and just calling `$note->set_status` etc directly.

Reply:
I think that what you said makes sense. What I was thinking when I started with this refactor was that probably this method [was created](https://github.com/woocommerce/woocommerce-admin/blob/master/src/API/Notes.php#L201) with the idea of allowing some specific changes and nothing else.
What I did here specifically was:
1. move some parts of the method from `Notes.php` to this file. 
2. add the change that I needed, trying to keep the same structure.

The refactor you mentioned sounds interesting, maybe we can create a specific issue to discuss the change to make it.

Reply:
Ok I understand. I didn't realise this was part of a refactor and was adding to existing code. This should be fine then :-)